lxd/storage/backend_lxd.go

```diff
@@ -2119,92 +2121,164 @@ func (b *lxdBackend) CreateInstanceFromMigration(inst instance.Instance, conn io
 
-			if imageExists {
-				l.Debug("Using optimised migration from existing image", logger.Ctx{"fingerprint": fingerprint})
+					// Populate the volume filler with the fingerprint and image filler
+					// function that can be used by the driver to pre-populate the
+					// volume with the contents of the image.
+					preFiller = drivers.VolumeFiller{
+						Fingerprint: fingerprint,
+						Fill:        b.imageFiller(fingerprint, op),
+					}
 
-				// Populate the volume filler with the fingerprint and image filler
-				// function that can be used by the driver to pre-populate the
-				// volume with the contents of the image.
-				preFiller = drivers.VolumeFiller{
-					Fingerprint: fingerprint,
-					Fill:        b.imageFiller(fingerprint, op),
+					// Ensure if the image doesn't yet exist on a driver which supports
+					// optimized storage, then it gets created first.
+					err = b.EnsureImage(preFiller.Fingerprint, op)
+					if err != nil {
+						return err
+					}
 				}
+			}
+		}
 
-				// Ensure if the image doesn't yet exist on a driver which supports
-				// optimized storage, then it gets created first.
-				err = b.EnsureImage(preFiller.Fingerprint, op)
-				if err != nil {
-					return err
-				}
+		// Retrieve a list of target volume snapshots.
+		// Afterwards load the volume from the snapshot to ensure the right ordering.
+		instSnapshots, err := inst.Snapshots()
+		if err != nil {
+			return err
+		}
+
+		targetSnapshots := make([]drivers.Volume, 0, len(instSnapshots))
+		for _, instSnapshot := range instSnapshots {
+			snap, err := VolumeDBGet(b, inst.Project().Name, instSnapshot.Name(), volType)
+			if err != nil {
+				return err
 			}
+
+			snapshotStorageName := project.Instance(inst.Project().Name, instSnapshot.Name())
+			targetSnapshots = append(targetSnapshots, b.GetVolume(volType, contentType, snapshotStorageName, snap.Config))
 		}
-	}
 
-	// Retrieve a list of target volume snapshots.
-	// Afterwards load the volume from the snapshot to ensure the right ordering.
-	instSnapshots, err := inst.Snapshots()
-	if err != nil {
-		return err
-	}
+		volCopy := drivers.NewVolumeCopy(vol, targetSnapshots...)
 
-	targetSnapshots := make([]drivers.Volume, 0, len(instSnapshots))
-	for _, instSnapshot := range instSnapshots {
-		snap, err := VolumeDBGet(b, inst.Project().Name, instSnapshot.Name(), volType)
+		err = b.driver.CreateVolumeFromMigration(volCopy, conn, args, &preFiller, op)
 		if err != nil {
 			return err
 		}
 
-		snapshotStorageName := project.Instance(inst.Project().Name, instSnapshot.Name())
-		targetSnapshots = append(targetSnapshots, b.GetVolume(volType, contentType, snapshotStorageName, snap.Config))
-	}
+		if !isRemoteClusterMove {
+			revert.Add(func() { _ = b.DeleteInstance(inst, op) })
+		}
+	} else {
+		// Get source image path (where image will be uploaded) from volume arguments.
+		srcImagePath := args.ConvertFilePath
```

Comment:
Actually, I have to introduce `conversionMode` in this PR, because otherwise users cannot avoid the unnecessary upload of the image to the backups directory.

Reply:
Maybe `conversionOpts` or `conversionArgs` is a more suitable name, since it will accept list of options. For example: `--conversion=format,virtio`

Reply:
im not too fussed either way. convertionOpts would be my preference.